lib/SimpleHttpClient/SslClientConnection.cpp

```diff
@@ -343,15 +379,45 @@ bool SslClientConnection::connectSocket() {
 
   ERR_clear_error();
 
-  int ret = SSL_connect(_ssl);
+  int ret = -1;
+  int errorDetail = -1;
+
+  if (_isSocketNonBlocking) {
+    auto start = steady_clock::now();
+    while ((ret = SSL_connect(_ssl)) == -1) {
+      errorDetail = SSL_get_error(_ssl, ret);
+      if (_isInterrupted) {
+        break;
+      }
+      auto end = steady_clock::now();
+      if ((errorDetail != SSL_ERROR_WANT_READ &&
+           errorDetail != SSL_ERROR_WANT_WRITE) ||
+          duration_cast<seconds>(end - start).count() >= _connectTimeout) {
+        break;
+      }
+      std::this_thread::sleep_for(seconds(1));
```

Comment:
Sleeping for 1s is too long, because we want to abort the loop relatively quickly if necessary.
What about 50ms instead?